### PR TITLE
Branch2 lukesalter71 viewfunction

### PR DESCRIPTION
**Pull request by lukesalter71**
Built a view function for taking quizzes. 

Currently I've left some code commented in main.py out since they rely on stuff that hasn't been implemented yet:
> a handler for a quiz results screen, to redirect to once a user has completed a quiz (submitted the final question)

> a "quiz.html" template to render for GET requests

### CODE REVIEW

File: website/main.py [hunk 1]
```diff
@@ -20,4 +21,39 @@ def courses():
...
+@login_required
+def quiz(quizid):
+    if session['question_number'] == -1:                                                                #Checking that all the pre-requisites for taking a quiz have been met.
+        session['question_number'] = 0 
+        session.modified = True
+    questions = db.Question_Bank.query.filter_by(id=quizid)                                             #extract the given quiz's questions from the question_bank table into an array.
+    question = questions[session['question_number']]
+    answers = []
+    for i in question.answers:                                                                          #extract the current question's answers from the database.
+        answers.append(i.answer_text)
```
Comment (Abdi-VGW) [inline website/main.py:35]: same with this comment

File: website/main.py [hunk 2]
```diff
@@ -20,4 +21,39 @@ def courses():
...
+    
+    #grab the form object:
+    question_form = forms.Question()
+
+    if question.question_type == 2:                                                                     #If the question is a multiple-choice question
+        question_form.choices = answers                                                                 #set the radio buttons to the question's choices
```
Comment (Abdi-VGW) [inline website/main.py:41]: same with this comment

File: website/main.py [hunk 3]
```diff
@@ -20,4 +21,39 @@ def courses():
...
+
+    if question_form.validate_on_submit():                                                              #POST Request
+
```
Comment (Abdi-VGW) [inline website/main.py:44]: same with this comment. The function already defines the requests through the use of `methods=['GET','POST']`

File: website/main.py [hunk 4]
```diff
@@ -20,4 +21,39 @@ def courses():
...
+        if question.question_type == 2:
+            user_answer = question_form.choices.data                                                    #takes whichever response the user chooses
+        elif question.question_type == 1:                                                               #not using else, as another category of answer may be added in the sites future.
+            user_answer = question_form.answer_text.data
+
+        #check if the answered question is the last in the quiz:
+        #if int(session['question_number'])+1 == len(questions):                                         # e.g. the 5th question on a 5-question quiz would have session['question_number'] = 4. Hence the +1.
+            #return(redirect(url_for('marks', quizid = quizid)))                                         #redirect to the marks screen (not ready yet, so commented this out.)
+
+        #since any quiz that has passed the final question would have redirected at this point, we can safely move to the next question
```
Comment (Abdi-VGW) [inline website/main.py:54]: I would remove all these comments as well as the code makes sense and is readable already

File: website/main.py [hunk 5]
```diff
@@ -20,4 +21,39 @@ def courses():
...
\ No newline at end of file
+#Quiz taking view function (todo)
+@main.route('/quiz/<quizid>', methods=['GET','POST'])
+@login_required
+def quiz(quizid):
+    if session['question_number'] == -1:                                                                #Checking that all the pre-requisites for taking a quiz have been met.
+        session['question_number'] = 0 
+        session.modified = True
+    questions = db.Question_Bank.query.filter_by(id=quizid)                                             #extract the given quiz's questions from the question_bank table into an array.
+    question = questions[session['question_number']]
```
Comment (Abdi-VGW) [inline website/main.py:32]: I would remove this comment since the code makes sense and is already very readable

**Review by Abdi-Isse — APPROVED** [review]
Nice

**Pull request merged**